src/gui/uithememanager.cpp

```diff
@@ -109,19 +130,20 @@ QIcon UIThemeManager::getIcon(const QString &iconId, const QString &fallback) co
 #endif
     // cache to avoid rescaling svg icons
     static QHash<QString, QIcon> iconCache;
-    const auto iter = iconCache.find(iconId);
+    const QString iconPath = getIconPath(iconId);
+    const auto iter = iconCache.find(iconPath);
     if (iter != iconCache.end())
         return *iter;
 
-    const QIcon icon {IconProvider::instance()->getIconPath(iconId)};
-    iconCache[iconId] = icon;
+    const QIcon icon {iconPath};
+    iconCache[iconPath] = icon;
     return icon;
 }
 
 QIcon UIThemeManager::getFlagIcon(const QString &countryIsoCode) const
 {
     if (countryIsoCode.isEmpty()) return {};
-    return QIcon(":/icons/flags/" + countryIsoCode.toLower() + ".svg");
+    return QIcon(m_iconsDir + "flags/" + countryIsoCode.toLower() + ".svg");
 }
 
 QString UIThemeManager::getIconPath(const QString &iconId) const
```

Comment:
Do the code inside `if (m_useSystemTheme) {}` still work? It is important for linux users to use native icons.

Reply:
>It is important for linux users to use native icons.

As I understand it, "native icons" should be used (if enabled) no matter what theme is selected, built-in or custom?

Reply:
>As I understand it, "native icons" should be used (if enabled) no matter what theme is selected, built-in or custom?

I think yes.

Reply:
Dor this we can have another config file with system icons as mapped value

Reply:
I think something like that will be needed...

Reply:
ok, I tried but didn't test. Actually, I don't have any Linux computer available and never used one either so it will take me some time to set up one. Can anybody else test this?